lib/cartopy/_proj4.py

```
@@ -0,0 +1,154 @@
+import shapely.geometry as sgeom
+
+_GLOBE_PARAMS = {'datum': 'datum',
+                 'ellps': 'ellipse',
+                 'a': 'semimajor_axis',
+                 'b': 'semiminor_axis',
+                 'f': 'flattening',
+                 'rf': 'inverse_flattening',
+                 'towgs84': 'towgs84',
+                 'nadgrids': 'nadgrids'}
+PROJ_TO_CRS = {}
+
+
+def get_proj4_dict(proj4_terms):
+    """Convert a PROJ.4 string to a dictionary.
+
+    Parameters
+    ----------
+    proj4_terms: (str, dict, or iterable of key-value pairs)
+
+    Returns
+    -------
+    get_proj4_dict
+        All PROJ.4 parameters in a dictionary. Any keys with no value
+        are set to `None`.
+
+    """
+    if isinstance(proj4_terms, dict):
+        return proj4_terms
+    elif isinstance(proj4_terms, str):
+        terms = []
+        for term in proj4_terms.split(' '):
+            parts = term.strip('+').split('=')
+            if len(parts) == 1:
+                terms.append((parts[0], None))
+            else:
+                terms.append(tuple(parts[:2]))
+    else:
+        # assume list of key value pairs
+        terms = proj4_terms
+
+    return dict(terms)
+
+
+def _split_globe_parameters(proj4_dict):
+    projection_terms = {}
+    globe_terms = {}
+    for name, value in proj4_dict.items():
+        if name in _GLOBE_PARAMS:
+            globe_terms[name] = value
+        else:
+            projection_terms[name] = value
+    return projection_terms, globe_terms
+
+
+def _globe_from_proj4(globe_terms):
+    """Create a `Globe` object from PROJ.4 parameters."""
+    globe = ccrs.Globe(**{_GLOBE_PARAMS[name]: value for name, value in
+                          globe_terms.items()})
+    return globe
+
+
+class _PROJ4Projection(ccrs.Projection):
+    def __init__(self, proj4_terms, globe=None, bounds=None):
+        terms = get_proj4_dict(proj4_terms)
+        projection_terms, globe_terms = _split_globe_parameters(terms)
+        if globe is None:
```

I think we need to raise an exception here is globe is not None and ``globe_terms`` is not empty.

---

I was following the base CRS class which uses `Globe()` as its default. I can add an exception, but with this default in mind, does that change your opinion?